### ISSUE TITLE
Support for underscore prefix

### ISSUE THREAD

**Issue by mertafor** [vscode-icons/vscode-icons#3348]
### Housekeeping

- [X] I'm sure this issue is _not_ a duplicate

### Information:

Next JS recommends using underscore prefix for folders such as hooks, components etc. in app folder. This might be a new common pattern for other frameworks as well. I know it's possible to add custom  icons but is there an easier way to catch folder names with underscore?

For example :
_hooks = hooks
_components = components
_lib = lib
_utils = utils

and so forth.

**Comment by ziyafenn** [comment]
Created an issue to add a complete Next.js folder pattern support: https://github.com/vscode-icons/vscode-icons/issues/3434

**Comment by remcohaszing** [comment]
Unfortunately the VSCode API only allows specifying icons for directory names, file names, file extensions, and language IDs. This means we can’t implement this in an extension.

See #932

**Comment by jurend** [comment]
How come that [Material Icon Theme](https://github.com/material-extensions/vscode-material-icon-theme) supports underscore prefixes?

**Comment by remcohaszing** [comment]
Interesting, I didn’t know they support it. I may have closed this too soon.

It’s not possible to support patterns. Also with a quick glance, I don’t see support for it in the Material Icon Theme repo. Yet I see them when enabled. I also see it for directory names where it doesn’t make sense IMO. I suspect they map over their list of folder names and tell VSCode to add an icon for example for `src` as well ad `_src` and `.src`. But this is also applied to folder names where it doesn’t really make sense IMO, for example `_node_modules` and `.node_modules`.

@robertohuertasm Is this something we want to support?

**Comment by robertohuertasm** [comment]
@remcohaszing you're right about they creating those special folder names by default. See [here](https://github.com/material-extensions/vscode-material-icon-theme/blob/5a653d98f2a1bb7ee7aae28a78c60ef21c911fd0/src/core/generator/folderGenerator.ts#L290).

I agree that it generates mappings that don't make sense. What we've been supporting is adding those variants when needed. I think I would stick to that convention. For instance, if we see that the `_hooks` folder is widely used, then we add it. 

As @mertafor mentioned, we have the ability to customize the icon associations, and they can be set up as a user level for users always defaulting to use any particular folder variants like `_lib`.
